flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1420,6 +1424,41 @@ private void timeoutSlot(AllocationID allocationId, UUID ticket) {
 		}
 	}
 
+	/**
+	 * Sync slot with job master.
+	 * The slots only exist in job master will be failed to job master.
+	 * The slots only exist in task executor will be released to resource manager.
+	 *
+	 * @param slotSnapshot is the snapshot of slots in job master
```

Comment:
`@param slotSnapshot represents the JobMaster's view on the current slot allocation state`

Reply:
Done